cardano-cli/src/Cardano/CLI/Legacy/Run/Transaction.hs

```diff
@@ -168,7 +169,7 @@ runLegacyTransactionCmds = \case
 runLegacyTransactionBuildCmd
   :: ()
   => SocketPath
-  -> EraInEon ShelleyBasedEra
+  -> Exp.Some Exp.Era
```

Comment:
We need to introduce a deprecation message for the legacy commands as well.

Reply:
I think you've started working on this: https://github.com/IntersectMBO/cardano-cli/pull/905 - so it's out of scope of this PR?

Reply:
This PR can be merged as is. I'll rebase mine on it.